modules/homa-lite/src/lib.rs

```diff
@@ -260,6 +260,12 @@ pub mod module {
 	#[pallet::getter(fn staking_interest_rate_per_update)]
 	pub type StakingInterestRatePerUpdate<T: Config> = StorageValue<_, Permill, ValueQuery>;
 
+	/// Next redeem request to iterate from when matching redeem requests.
+	/// NextRedeemRequestToMatch: Value: T::AccountId
+	#[pallet::storage]
+	#[pallet::getter(fn next_redeem_request_to_match)]
+	pub type NextRedeemRequestToMatch<T: Config> = StorageValue<_, T::AccountId, ValueQuery>;
```

Comment:
this should store the raw key instead of AccountId. otherwise if this account request is redeemed or cancelled, the match will start from beginning again.

Reply:
Changed the storage to use key instead.
This storage is change to store the last redeem request key, since iter_from exclude from the input key